packages/cli/src/controllers/__tests__/auth.controller.test.ts

```diff
@@ -0,0 +1,99 @@
+import type { LoginRequestDto } from '@n8n/api-types';
+import { Container } from '@n8n/di';
+import type { Response } from 'express';
+import { mock } from 'jest-mock-extended';
+import { Logger } from 'n8n-core';
+
+import * as auth from '@/auth';
+import { AuthService } from '@/auth/auth.service';
+import config from '@/config';
+import type { User } from '@/databases/entities/user';
+import { UserRepository } from '@/databases/repositories/user.repository';
+import { EventService } from '@/events/event.service';
+import { License } from '@/license';
+import { MfaService } from '@/mfa/mfa.service';
+import { PostHogClient } from '@/posthog';
+import type { AuthenticatedRequest } from '@/requests';
+import { UserService } from '@/services/user.service';
+import { mockInstance } from '@test/mocking';
+
+import { AuthController } from '../auth.controller';
+
+jest.mock('@/auth');
+
+const mockedAuth = auth as jest.Mocked<typeof auth>;
+
+describe('AuthController', () => {
+	mockInstance(Logger);
+	mockInstance(EventService);
+	mockInstance(AuthService);
+	mockInstance(MfaService);
+	mockInstance(UserService);
+	mockInstance(UserRepository);
+	mockInstance(PostHogClient);
+	mockInstance(License);
+	const controller = Container.get(AuthController);
+	const userService = Container.get(UserService);
+	const authService = Container.get(AuthService);
+	const eventsService = Container.get(EventService);
+	const postHog = Container.get(PostHogClient);
+
+	describe('login', () => {
+		it('should not validate email in "emailOrLdapUsername if LDAP is enabled"', async () => {
```

Comment:
I'm not clear on the test here. Can't it be either an email or a username for LDAP?

Reply:
this tests check that if a LDAP is enabled do not validate that "emailOrLdapUsername" is an email